src/diffusers/pipelines/pipeline_loading_utils.py

```diff
@@ -475,6 +601,12 @@ def load_sub_model(
         # else load from the root directory
         loaded_sub_model = load_method(cached_folder, **loading_kwargs)
 
+    if isinstance(loaded_sub_model, torch.nn.Module) and device_map_strategy == "balanced":
+        if needs_offloading:
```

Comment:
As said above, let's use the check `device_map == "cpu" `instead.

Reply:
You mean instead of `needs_offloading` I check if "cpu" is present in the current `device_map`?